synapse/handlers/room_summary.py

```diff
@@ -1096,9 +1096,48 @@ async def get_room_summary(
 
                 room_summary["membership"] = membership or "leave"
         else:
-            # TODO federation API, descoped from initial unstable implementation
-            #      as MSC needs more maturing on that side.
-            raise SynapseError(400, "Federation is not currently supported.")
+            # Reuse the hierarchy query over federation
+            if remote_room_hosts is None:
+                raise SynapseError(400, "Missing via to query remote room")
+
+            fed_rooms = await self._summarize_remote_room(
```

Comment:
`_summarize_remote_room` seems to use `/spaces/<room_id>` which I can't find in the final version of msc2946.
Should we be using this endpoint?
For reference, `_summarize_remote_room_hierarchy` uses `/hierarchy/<room_id>`